src/js/components/user/githubUser.tsx

```diff
@@ -0,0 +1,263 @@
+
+  // When modal is canceled, reset row selection
+  const handleClose = () => {
+    reset();
+    onRequestClose();
+  };
+  const updateSelection = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    data: { selection: GitHubUser[] },
+  ) => {
+    setSelection(data.selection);
+  };
+  const handleSubmit = () => {
+    setUsers([...selection]);
+    reset();
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      heading={heading}
+      footer={[
+        <Button key="cancel" label={i18n.t('Cancel')} onClick={handleClose} />,
+        <Button
+          key="submit"
+          type="submit"
+          label={i18n.t('Save')}
+          variant="brand"
+          onClick={handleSubmit}
+        />,
+      ]}
+      onRequestClose={handleClose}
+    >
+      <DataTable
+        className="align-checkboxes"
+        items={allUsers}
+        selectRows="checkbox"
+        selection={selection}
+        onRowChange={updateSelection}
+        noRowHover
+      >
+        <DataTableColumn
+          label={i18n.t('GitHub Username')}
+          property="login"
+          primaryColumn
+          truncate
+        >
+          <UserTableCell />
+        </DataTableColumn>
+      </DataTable>
+    </Modal>
+  );
+};
+
+const GitHubUserButton = ({
+  user,
+  isSelected,
+  ...props
+}: {
+  user: GitHubUser;
+  isSelected?: boolean;
+  [key: string]: any;
+}) => (
+  <Button
+    className={classNames(
+      'slds-size_full',
+      'slds-p-around_xx-small',
+      'collaborator-button',
+      {
+        'is-selected': isSelected,
+      },
+    )}
+    title={user.login}
+    label={
+      <>
+        <GitHubUserAvatar user={user} />
+        <span className="collaborator-username">{user.login}</span>
+      </>
+    }
+    variant="base"
+    disabled={isSelected}
+    {...props}
+  />
+);
+
+export const AssignUserModal = ({
+  allUsers,
+  selectedUser,
+  heading,
+  isOpen,
+  emptyMessageAction,
+  onRequestClose,
+  setUser,
+}: {
+  allUsers: GitHubUser[];
+  selectedUser: GitHubUser | null;
+  heading: string;
+  isOpen: boolean;
+  emptyMessageAction: () => void;
+  onRequestClose: () => void;
+  setUser: (user: GitHubUser | null) => void;
+}) => {
+  const filteredUsers = allUsers.filter((user) => user.id !== selectedUser?.id);
+
+  return (
+    <Modal isOpen={isOpen} heading={heading} onRequestClose={onRequestClose}>
+      {selectedUser && (
+        <>
+          <div className="slds-p-around_small">
+            <div className="slds-text-title slds-m-bottom_xx-small">
+              {i18n.t('Currently Assigned')}
+            </div>
+            <GitHubUserButton user={selectedUser} isSelected />
+          </div>
+          <hr className="slds-m-vertical_none slds-m-horizontal_small" />
+        </>
+      )}
+      {filteredUsers.length ? (
+        <div className="slds-p-around_small">
+          <div className="slds-text-title slds-m-bottom_xx-small">
+            {i18n.t('Assign To User')}
```

Comment:
Should we use the label collaborator here? Assign to  `developer` (or reviewer`) ... or generically "Change assignment to:"

Reply:
I changed to "Assign to GitHub User" so that it's clear these are GitHub usernames. Does that seem better?